testing/hooks_test.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+import { afterEach, beforeEach, withHooks } from "./hooks.ts";
+import { assert, assertEquals, AssertionError } from "./asserts.ts";
+
+Deno.test("beforeEach is exectued correctly", function (): void {
+  let didInvoke = false;
+  beforeEach(function () {
+    didInvoke = true;
+    assertEquals(/deno/, /deno/);
+  });
+  assertEquals(didInvoke, false);
+  withHooks(function () {
+    assertEquals(/cat/, /cat/);
+  })();
+  assertEquals(didInvoke, true);
+});
+
+Deno.test("afterEach is exectued correctly", function (): void {
+  let didInvoke = false;
+  afterEach(function () {
+    try {
+      assertEquals(/deno/, /cat/);
+    } catch (e) {
+      assert(e instanceof AssertionError);
+      didInvoke = true;
+    }
```

Comment:
This try-catch structure looks meaningless to me. It can be replaced by simply `didInvoke = true`. And we should check didInvoke value after `withHooks(...)` block to check wether it's actually called 'after' the main function.

Reply:
yes, i will remove try-catch structure